release process

juliev0 opened the pull request:
<!-- markdownlint-disable MD041 -->

<!-- Does this PR fix an issue -->

Fixes #212 

### Modifications

Mostly I imitated what numaflow is doing as part of its release process.

The release process only works on a branch prefixed by "release-" per [these](https://numaflow.numaproj.io/development/releasing/) release instructions. 

Ultimately we need to incorporate the pushing of the images into a CI release workflow (added [Issue](https://github.com/numaproj-labs/numaplane/issues/234)), so for now we can do it manually:
 
Run:
```
export VERSION=vx.y.z
make docker-buildx
make prepare-release
make release
```

It updates the manifests to use the new image tag, runs `codegen` target to make sure all of the auto-generated code is up to date, pushes all of this to the currently `release-x` branch, creates the tag based on $VERSION, and pushes it. 

Example: 
Current branch is `release-0.5`.
$VERSION is set to `v0.5.3`, which is what the git tag is set to. 
At the end of this both the release branch and the tag contain the latest and greatest.

### Verification

I started from this branch since it had these release process changes in it, and I created a new branch from it called `release-0.1`. Then I ran:
```
export VERSION=v0.1.0
make docker-buildx
make prepare-release
make release
```

You can see the branch [here has the correct image tag](https://github.com/numaproj-labs/numaplane/blob/release-0.1/config/install.yaml#L458).
You can see the [v0.1.0 tag has the correct image tag](https://github.com/numaproj-labs/numaplane/blob/v0.1.0/config/install.yaml#L458).
You can also see that the v0.1.0 tag exists for the image [here](https://quay.io/repository/numaio/numaplane-controller?tab=tags).

xdevxy commented on config/install.yaml:
```diff
@@ -455,7 +455,7 @@ spec:
         command:
         - /manager
-        image: quay.io/numaproj/numaplane-controller:latest
+        image: quay.io/numaproj/numaplane-controller:v0.1.0-beta.3
```
This tag change has to be merged?

juliev0 replied:
sorry, that's a mistake when I was testing

juliev0 commented on README.md:
````diff
@@ -9,58 +9,22 @@ Numaplane is a control plane for install, manage and run numaflow pipelines on K
-**Build and push your image to the location specified by `IMAGE_FULL_PATH`:**
+### To build Numaplane image and run it on your local cluster with latest manifests
 
-```sh
````
These instructions seemed to be outdated

juliev0 commented:
@xdevxy I've addressed your comments, updated the README.md, and also added a separate [issue](https://github.com/numaproj-labs/numaplane/issues/234) for updating the CI, which is the part that still needs to be done.
